tests/circuit_graph/test_circuit_graph.py

```diff
@@ -93,6 +93,40 @@ def circuit_measure_multiple_with_max_twice():
     )
 
 
+def measurement_process_equality(mp1, mp2):
+    """Helper function to check for equality between two MeasurementProcess objects."""
+    same_return_type = mp1.return_type == mp2.return_type
+    same_obs = mp1.obs == mp2.obs
+    same_wires = mp1._wires == mp2._wires
+    same_eigs = mp1._eigvals == mp2._eigvals
+    same_name = mp1.name == mp2.name
+    same_data = mp1.data == mp2.data
+
+    if same_return_type and same_obs and same_wires and same_eigs and same_name and same_data:
+        return True
+
+    return False
+
+
+def lst_mp_equality(lst1, lst2):
+    """Wrapper function to check if two lists are identical when the
+    lists contain MeasurementProcess objects."""
+    assert len(lst1) == len(lst2)
+    lsts_same = True
+
+    for index in range(len(lst1)):
+        if not isinstance(lst1[index], type(lst2[index])):  # if they are different types
+            return False  # then return false
+
+        elif isinstance(lst1[index], qml.measure.MeasurementProcess):
+            lsts_same = lsts_same and measurement_process_equality(lst1[index], lst2[index])
+
+        else:
+            lsts_same = lsts_same and (lst1[index] == lst2[index])
+
+    return lsts_same
```

Comment:
I added a test to ensure that the line of code I added would actually generate a circuit graph object such that the object generated by `qml.sample()` is identical to the one generated when `qml.sample(wires=[0,1,2])` with all the wires explicitly mentioned. 

When adding this implementation I came across a little problem: 
`obj1 = qml.measure.sample()` 
`obj2 = qml.measure.sample()`
`obj1 == obj2  #returns false` 

This is probably because the equality operator is checking if they are references to the same object as opposed to comparing their fields. For this reason I had to implement two other helper functions in the text file to allow me to more easily compare these MeasurementProcess objects without compromising code readability. 

I tried over-writing the `==` operator in the MeasurementProcess class, but this lead to other errors in other files, so I decide to go with this solution. Let me know what you think! 

Thanks,

Reply:
Thanks @Jaybsoni! I imagine defining `__eq__` for the `MeasurementProcess` class led to some issues with hashing of the object within the codebase 🤔 Interesting.

For now, this is a decent solution!